fabfile.py

```diff
@@ -843,6 +853,67 @@ def updatemaincode(backup=False):
 
             venvcmd_py3('pip install -r ../url_metadata/requirements.txt')
 
+        execute(update_bluenove_actionable)
+
+
+def get_robot_machine():
+    """
+    Return the configured robot machine: (the first configured machine)
+    """
+    machines = env.get('machines', '').split('/')
+    if machines:
+        robot = machines[0]
+        robot_data = robot.split(',')
+        return {
+            'identifier': robot_data[0].strip(),
+            'name': robot_data[1].strip(),
+            'password': robot_data[2].strip()
+        }
+
+    return None
+
+
+@task
+def update_bluenove_actionable():
+    path = join(env.projectpath, '..', 'bluenove-actionable')
+    if exists(path):
+        print(cyan('Updating bluenove-actionable Git repository'))
+        with cd(path):
+            # TODO We need an ssh access
+            run('git pull', warn_only=True)
+            execute(stop_bluenove_actionable)
+            run('docker-compose build --no-cache', warn_only=True)
+
+
+@task
+def stop_bluenove_actionable():
+    path = join(env.projectpath, '..', 'bluenove-actionable')
+    if exists(path):
+        print(cyan('stop bluenove-actionable'))
+        with cd(path):
+            fabsudo('docker kill bluenoveact', warn_only=True)
```

Comment:
yeah use `sudo` and not `fabsudo` here

Reply:
`docker kill` is hard, you should try to stop first. Better, use docker-compose too here. `docker-compose down` is what you want, it will stop, kill if necessary and remove the container.